python/ray/ray_operator/operator.py

```diff
@@ -141,23 +158,20 @@ def cluster_action(event_type, cluster_cr, cluster_name) -> None:
 
 
 def main() -> None:
-    # Make directory for ray cluster configs
+    # Make directory for Ray cluster configs
     if not os.path.isdir(operator_utils.RAY_CONFIG_DIR):
         os.mkdir(operator_utils.RAY_CONFIG_DIR)
-    # Control loop
-    cluster_cr_stream = operator_utils.cluster_cr_stream()
-    try:
-        for event in cluster_cr_stream:
-            cluster_cr = event["object"]
-            cluster_name = cluster_cr["metadata"]["name"]
-            event_type = event["type"]
-            handle_event(event_type, cluster_cr, cluster_name)
-    except ApiException as e:
-        if e.status == 404:
-            raise Exception(
-                "Caught a 404 error. Has the RayCluster CRD been created?")
-        else:
-            raise
+    while True:
+        try:
+            run_event_loop()
+        except ApiException as e:
+            # Wait for creation of the Ray Cluster CRD if it hasn't
+            # already been created.
+            if e.status == 404:
+                logger.warning("Waiting for creation of the RayCluster CRD")
+                time.sleep(5)
+            else:
+                raise
```

Comment:
what happens after this?

Reply:
That's an unhandled exception...
It would mean we failed to enter the event loop for some unforeseen reason.
The container crashes within a short period of starting, then restarts. 

If we got to this line of code on the first try, then probably we'll quickly get to it on the second try too. Eventually we'd get CrashLoopBackOff and could check what's wrong with kubectl logs.

Reply:
Umm, same behavior if the exception raised is not an ApiException.

Reply:
okay crash looping makes sense.